src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -467,6 +469,32 @@ protected virtual void SweepMethods (Collection<MethodDefinition> methods)
 			}
 		}
 
+		void SweepOverrides (MethodDefinition method)
+		{
+			for (int i = 0; i < method.Overrides.Count;) {
+				// We can't rely on the context resolution cache anymore, since it may remember methods which are already removed
+				// So call the direct Resolve here and avoid the cache.
+				// It can still happen that the Resolve will return a removed method (this happens if the Overrides collection stores actual
+				// MethodDefinition and not MethodReference, in which case the Resolve is just "return this" and doesn't check the existence of the method
+				// in any way). In such case the Declaring type of the method is null. And since it has been removed the override pointing
+				// to it should be removed as well.
+				// Resolve() may also return null if the method has been removed, in which case the override should be removed from the list
+				if (method.Overrides[i].Resolve () is not MethodDefinition ov || ShouldRemove (ov) && (ov.DeclaringType == null || !IgnoreScope (ov.DeclaringType.Scope)))
```

Comment:
Nit: Could you also explain in the comment about the ShouldRemove and the IgnoreScope?
Basically I think we should list all reasons a method override should be removed in an 'easy to understand list' - basically this but expanded:
* Doesn't resolve (couple of ways this can occur)
* Not marked (ShouldRemove), but only if coming from trimmed scope

I know that ideally the code should speak for itself, but this has proven to be really tricky, so having a good completely comment won't hurt.

Reply:
I've updated the comment here, let me know if it's clear and understandable.